stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -744,6 +746,7 @@ PollResult TaskGroupImpl::poll(AsyncTask *waitingTask) {
           result.status = PollStatus::Success;
           result.storage = futureFragment->getStoragePtr();
           assert(result.retainedTask && "polled a task, it must be not null");
+          _swift_tsan_acquire(static_cast<Job *>(result.retainedTask));
           mutex.unlock(); // TODO: remove fragment lock, and use status for synchronization
```

Comment:
Konrad also pointed out that the current implementation uses a lock (pending for removal), which of course adds additional edges for TSan.  This may explain my futile attempts of producing a test that "goes red".